src/js/components/FormField/FormField.js

```diff
@@ -451,7 +459,22 @@ const FormField = forwardRef(
         {...outerProps}
         style={outerStyle}
         onFocus={(event) => {
-          setFocus(shouldKeepFocus());
+          setFocus(containsFocus(formFieldRef.current));
+          const formDescendants =
+            formFieldRef.current.getElementsByTagName('*');
+          let focusableDescendants = 0;
+          for (let i = 0; i < formDescendants.length; i += 1) {
+            const child = formDescendants[i];
+            if (isFocusable(child)) {
+              focusableDescendants += 1;
+            }
+            // FormField descendant could be focusable. If descendant is
+            // active element in DOM, don't show focus on FormField.
+            // e.g. DateInput allows Calendar button to receive focus
+            if (focusableDescendants >= 2 && child === document.activeElement) {
```

Comment:
The activeElement check weeds out things that may not actually be visible (e.g. `display:none`) or may be `disabled`?

Reply:
Yes, if an element is disabled or if it has `visibility: 'hidden'` it will still show up in the `focusableDescendants` list but the `child === document.activeElement` check makes sure the focus isn't incorrectly removed from the FormField if it contains a disabled or invisible child. Focus will only be removed from the FormField if there is another focusable descendant and that element has focus

Reply:
Thanks. Sounds good.